### PR TITLE
feat(ci): flag Cli releases missing the Extension bundled-CLI refresh (closes #1359)

### LINKED ISSUE #1359: workflow-tests path filter misses .github/workflows/**; sync MERGE-POLICY tooling list
## Context

#1355 (closes #1353) runs the full `tests/` Python suite on pull_request via `.github/workflows/workflow-tests.yml`. Pre-release review found the guard has a self-blind spot, plus a doc-sync item in the policy it protects.

## Gap 1 — path filter misses `.github/workflows/**` ([80])

`workflow-tests.yml:26-31` triggers on `tests/**`, `scripts/**`, `.claude/**`, `CLAUDE.md`, and itself — but the suite asserts invariants of OTHER workflow files (`tests/ci/test_docs_release_workflow.py:20`, `test_extension_publish_workflow.py:17`, `test_post_merge_release_check.py:24`). A PR editing only `docs-release.yml` never triggers the guard; the violation merges green and the suite goes red later on an innocent PR touching `tests/**` — misattributed failure. Fix: add `.github/workflows/**` to the paths filter.

## Gap 2 — MERGE-POLICY tooling list drift ([40], cosmetic)

`docs/MERGE-POLICY.md:19` example tooling list "(eslint, knip, vitest, esbuild, @types/*, @playwright/test, Microsoft.NET.Test.Sdk)" predates #1351's classifier additions (`@vscode/test-electron`, `microsoft.codeanalysis.*` analyzers, `xunit.skippablefact`, `eslint-plugin-` / `stylelint-config-` prefixes). `classify.py` states the doc wins on disagreement — sync the doc's example list (or soften it to "see classify.py's allowlist").

## Optional (repo-owner call, noting for the record)

`workflow-tests` is not a required status check (branch protection requires only `build-status`/`test-status`), so its red X is advisory on approve-and-merge. Pre-existing posture, not introduced by #1355.

## Refs

#1355, #1353, #1351. Provenance: pre-release verification round (orchestrated multi-agent review), main @ 7d476550c.

### PR DESCRIPTION
**Pull request by joshsmithxrm**
## Incident

`Cli-v1.3.0` + `Mcp-v1.1.0` shipped 2026-07-14 **without an Extension release**. The VS Code Extension bundles the CLI at publish time (`extension-publish.yml` → `npm run bundle:cli`), so marketplace users kept running ~1.2.0-era CLI behavior — including a bug 1.3.0 fixes — because no `Extension-v*` refresh rode along to re-bundle the new CLI.

## The rule

**A `Cli-v*` release includes an `Extension-v*` bundled-CLI refresh in the same train — opt-out only with a recorded reason.** This PR is the *mechanical* backstop for that rule. The human-facing runbook paragraph is documented here too (see below) because the in-flight `release: Extension 1.4.1 (bundled CLI 1.3.0 refresh)` PR that was to carry it never landed.

## How the guard works

A new `extension-corelease-detection` job in `.github/workflows/post-merge-release-check.yml`:

- **Triggers** on a `Cli-v*` **tag push** (plus `workflow_dispatch` for a manual re-check). The existing job triggers on `pull_request: closed`, which can't see a tag push, so the triggers are extended minimally with `push.tags: ['Cli-v*']`. The two jobs are mutually exclusive by event (`github.event_name`), so neither cross-fires.
- **Compares** the latest **stable** `Cli-v*` tag against the latest `Extension-v*` tag via `scripts/ci/check_extension_corelease.py`. Because Cli and Extension version numbers are independent lines (Cli `1.3.0` vs Extension `1.4.1`), "is the Extension stale?" is answered by **tag date**, not version number: if the newest Extension tag was created *before* the newest stable Cli tag, the refresh is missing.
  - **Stable filter:** `-rc.`/`-beta.` suffixed Cli tags are excluded (a prerelease CLI never ships to marketplace users). Extension channels use the odd/even-minor convention, not suffixes, and *either* channel bundles a fresh CLI — so the comparison deliberately treats any newer Extension tag as a satisfied refresh (well-commented in the script).
- **Files** an issue titled exactly `Extension bundled-CLI refresh missing for Cli-vX.Y.Z (latest Extension-vA.B.C predates it)` through the workflow's existing `gh issue create` mechanism, with a release checklist.
- **Dedups** by reusing the cadence check's label pattern: it counts open `release:extension-corelease` issues first and passes `--has-open-issue`, so repeated runs never file a duplicate. The label is upserted idempotently (`gh label create --force`) since it's new to the repo.

## Guard the guard (Closes #1359)

- **`workflow-tests.yml` path filter** now includes `.github/workflows/**` (both `pull_request` and `push`). Previously a PR editing only a workflow never triggered the suite that asserts workflow invariants (including the new co-release tests) — the violation would merge green and surface as a misattributed red X on an unrelated PR. This was #1359 Gap 1.
- **`docs/MERGE-POLICY.md` tooling list** synced with `scripts/dependabot/classify.py`'s current `TOOLING_PACKAGES`/`TOOLING_PREFIXES` allowlist and pointed at the classifier as authoritative (it had drifted post-#1351). This was #1359 Gap 2.

## Tests

`tests/ci/test_extension_corelease_check.py` (20 tests), in the existing test file's style:

- **(a)** Cli tag newer than Extension → flagged, asserting the **exact** message.
- **(b)** Extension tag current or newer → silent (including same-instant same-train).
- **(c)** prerelease Cli tags ignored — a newer `-rc.` tag does not mask the last stable release.
- Plus dedup, missing-tag edge cases, version-rank selection, issue-body content, and workflow wiring (tag-push trigger + script/label presence).

Each fails without the change (the test module imports the new script; workflow-wiring tests assert the new trigger/label).

## Gates

- `python -m pytest tests/ scripts/ --import-mode=importlib -q` → **551 passed**.
- Both touched workflow YAMLs validated with `yaml.safe_load`.

## Sweep results

| Item | Disposition |
|---|---|
| 1.4.1 release PR / runbook paragraph | **Never landed** (no PR exists). Runbook paragraph added to `.claude/skills/release/REFERENCE.md` §1 — not `SKILL.md`, which is 626 lines and blocked by the 150-line-cap hook; the repo's two-file pattern puts runbook detail in REFERENCE.md. |
| #1359 Gap 1 (path filter) | Fixed |
| #1359 Gap 2 (MERGE-POLICY drift) | Fixed |

Closes #1359.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

- **New Features**
  - Added an automated co-release guard to detect when a new CLI stable release is missing the corresponding Extension bundled refresh, using tag timing and stable-only selection.
  - Creates a deduplicated follow-up issue with standardized messaging; stays quiet when the Extension refresh is current.

- **CI / Workflows**
  - Added a tag-push trigger so the guard runs on new matching CLI tags (with manual re-runs).
  - Broadened workflow regression coverage to run when any workflow definition changes.

- **Documentation**
  - Updated the release reference with the default co-release rule and opt-out requirements.
  - Expanded Dependabot auto-merge eligibility examples and pointed to the authoritative allowlist.

- **Tests**
  - Added unit tests covering detection, prerelease exclusion, idempotency, and issue/workflow wiring.

### CODE REVIEW

**Comment by joshsmithxrm** [comment]
Thanks @coderabbitai — both addressed in 4a0a17105:
- `collect_tags` now sets `encoding='utf-8'`/`errors='replace'` and **raises** on a non-zero git exit, so a git failure fails the CI step loudly rather than masquerading as "no tags" (which would silently suppress the guard).
- `main` now skips `--format title`/`body` output when there's no stable `Cli-v*` tag, so the standalone path can never emit a message naming a `None` tag. (The workflow already only requests title/body when `should_open_issue == true`, where `cli_tag` is non-null.)

**Comment by joshsmithxrm** [comment]
Addressed in d12e2786d: reframed the MERGE-POLICY tooling bullet as **examples-only** and pointed readers to `is_tooling_package()` / `TOOLING_PACKAGES` / `TOOLING_PREFIXES` in `scripts/dependabot/classify.py` as the source of truth — this avoids re-drift the next time the classifier's allowlist changes, rather than enumerating an exhaustive list that would go stale again.

File: scripts/ci/check_extension_corelease.py [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+    result = subprocess.run(
+        [
+            "git",
+            "for-each-ref",
+            "--format=%(refname:short)%09%(creatordate:iso-strict)",
+            f"refs/tags/{prefix}-v*",
+        ],
+        capture_output=True,
+        text=True,
+    )
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply scripts/ci/check_extension_corelease.py:266]: Addressed in 4a0a17105. `collect_tags` now sets `encoding='utf-8'`/`errors='replace'`. On a non-zero git exit it raises `RuntimeError` (surfacing `stderr`) so a git failure fails the CI step loudly rather than masquerading as "no tags" — which would silently suppress the guard and file/skip an issue on bad data.

File: scripts/ci/check_extension_corelease.py [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+        cli=cli,
+        extension=extension,
+        has_open_issue=args.has_open_issue,
+    )
+
+    if args.format == "title":
+        # Only meaningful when a flag is warranted; callers gate on should_open_issue.
+        print(build_flag_message(result["cli_tag"], result["extension_tag"]))
+    elif args.format == "body":
+        print(build_issue_body(result["cli_tag"], result["extension_tag"]))
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply scripts/ci/check_extension_corelease.py]: Addressed in 4a0a17105. `main` now guards the `title`/`body` branches: when `result["cli_tag"]` is `None` it prints a note to stderr and returns without emitting output, so no message ever names a `None` tag. (The workflow only requests title/body when `should_open_issue == true`, where `cli_tag` is guaranteed non-null.)
